### PR TITLE
add branch location priority to location update

### PR DESCRIPTION
**Pull request by rythod**

### CODE REVIEW

File: spec/requests/branch_request_spec.rb [hunk 1]
```diff
@@ -433,6 +433,16 @@
     context 'with branch owner' do
       let(:token) { owner_token }
 
+      describe 'when updating priority' do
+        let(:params) do 
+          { priority: rand(-1..10) }
```
Comment (gbudiman) [inline spec/requests/branch_request_spec.rb:438]: This test has 2 problems.
- Priority defaults to 0. There is a 1/12 chance that this test seeds a params with priority 0, thus nullifying the test.
- Similarly, there is only 1/12 chance the test seeds params with negative value. In other words, we're only testing negative values 1/12 of the time.

Solution:
Just make 2 distinct `describe` block. One accepts negative priority params, others positive priority params. And avoid priority 0.

In general, avoid using `rand()` in test. That is the source of non-deterministic failure (a.k.a Schrödinger's Cat problem).

File: app/controllers/branches_controller.rb [hunk 2]
```diff
@@ -104,7 +104,7 @@ def branch_update_params
   end
 
   def branch_location_params
-    params.permit(:name, :street_address, :city, :region, :zip_code)
+    params.permit(:name, :street_address, :city, :region, :zip_code, :priority)
```
Comment (gbudiman) [inline app/controllers/branches_controller.rb:107]: The priority should be in location controller

File: app/controllers/locations_controller.rb [hunk 3]
```diff
@@ -6,8 +6,10 @@ class LocationsController < ApplicationController
 
   def update
     render_as_json do
-      location.update!(update_params)
-      location.as_json(location_as_json)
+      location.update!(update_params.except(:priority))
+      branch_location.update!(params.permit(:priority)) if params[:priority].present?
```
Comment (gbudiman) [inline app/controllers/locations_controller.rb:10]: This should be
`bl.update!(update_params[:priority]) if update_params[:priority].present?`

Comment (gbudiman) [reply app/controllers/locations_controller.rb:10]: In general, except for `:*id`, do not access `params` directly.

File: app/controllers/branches_controller.rb (hunk 2 above)
Comment (rythod) [reply app/controllers/branches_controller.rb]: ok fixed

**Pull request merged**